src/frontend/src/handler/create_source.rs

```diff
@@ -113,20 +119,43 @@ pub async fn handle_create_source(
 
     let mut with_properties = context.with_options.inner().clone();
 
+    // pass `use_schema_registry` to `SourceParserImpl::create` by insert it into `with_properties`
+    const USE_SCHEMA_REGISTRY: &str = "use_schema_registry";
+    const UPSTREAM_SOURCE_KEY: &str = "connector";
+
     let (columns, source_info) = match &stmt.source_schema {
         SourceSchema::Protobuf(protobuf_schema) => {
             // the key is identified with SourceParserImpl::create
             const PROTOBUF_MESSAGE_KEY: &str = "proto.message";
 
+            // confluent schema registry must be used with kafka
+            if protobuf_schema.use_schema_registry
+                && !with_properties
+                    .get(UPSTREAM_SOURCE_KEY)
+                    .unwrap_or(&"".to_string())
+                    .to_lowercase()
+                    .eq("kafka")
+            {
+                return Err(RwError::from(ProtocolError(format!(
+                    "Must specify '{}' in WITH clause",
+                    UPSTREAM_SOURCE_KEY
+                ))));
+            }
+
             assert_eq!(columns.len(), 1);
             assert_eq!(pk_column_ids, vec![0.into()]);
             assert_eq!(row_id_index, Some(0));
 
             // unlike other formats, there are multiple messages in one file. Will insert a key to
             // identify the desired message.
-            with_properties
-                .entry(PROTOBUF_MESSAGE_KEY.into())
-                .or_insert_with(|| protobuf_schema.message_name.0.clone());
+            with_properties.insert(
+                PROTOBUF_MESSAGE_KEY.to_owned(),
+                protobuf_schema.message_name.0.clone(),
+            );
+            with_properties.insert(
+                USE_SCHEMA_REGISTRY.to_owned(),
+                protobuf_schema.use_schema_registry.to_string(),
+            );
```

Comment:
pass the info proto

Reply:
Shoud we add both `PROTOBUF_MESSAGE_KEY ` and `USE_SCHEMA_REGISTRY ` into `StreamSourceInfo ` or only `USE_SCHEMA_REGISTRY `.

Reply:
I suggest both